mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveShell.java

```diff
@@ -79,7 +79,7 @@ public void setHiveSessionValue(String key, boolean value) {
 
   public void start() {
     // Create a copy of the HiveConf for the metastore
-    metastore.start(new HiveConf(hs2Conf));
+    metastore.start(new HiveConf(hs2Conf), 10);
```

Comment:
Do we know why the pool is exhausted? In the past, we had a few leaks in the Spark catalog code which led to this. It can be also a valid use case too if we simply need a larger pool.

Reply:
Maybe, worth creating an issue if we want to take a look later.

Reply:
Or is it related to the recent change around how we manage the pool?

Reply:
I have 2 ideas:
- Maybe this was caused by the #2129 where we start to handle / cache the catalogs differently
- Or we added some more queries and Hive creates / caches HMSClients and we might ended up using a different codepath.

Created #2474